From 5273f61c9e93d1fa263fba29f0d62ab2860cde65 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 21 May 2016 21:52:36 -0700
Subject: [PATCH 1/4] clang: driver: Use /lib for ldso on OE

OE does not follow the default base_libdir
that clang has, therefore adjust it for OE
it wont be able to support multilib since
in multilib case OE switches the base libdir
for 64bit to /lib64 instead of /lib

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 lib/Driver/ToolChains.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 7884c40..706612d 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -4227,12 +4227,12 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     Loader = "ld.so.1";
     break;
   case llvm::Triple::ppc64:
-    LibDir = "lib64";
+    LibDir = "lib";
     Loader =
         (tools::ppc::hasPPCAbiArg(Args, "elfv2")) ? "ld64.so.2" : "ld64.so.1";
     break;
   case llvm::Triple::ppc64le:
-    LibDir = "lib64";
+    LibDir = "lib";
     Loader =
         (tools::ppc::hasPPCAbiArg(Args, "elfv1")) ? "ld64.so.1" : "ld64.so.2";
     break;
@@ -4242,7 +4242,7 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     Loader = "ld-linux.so.2";
     break;
   case llvm::Triple::sparcv9:
-    LibDir = "lib64";
+    LibDir = "lib";
     Loader = "ld-linux.so.2";
     break;
   case llvm::Triple::systemz:
@@ -4256,7 +4256,7 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
   case llvm::Triple::x86_64: {
     bool X32 = Triple.getEnvironment() == llvm::Triple::GNUX32;
 
-    LibDir = X32 ? "libx32" : "lib64";
+    LibDir = "lib";
     Loader = X32 ? "ld-linux-x32.so.2" : "ld-linux-x86-64.so.2";
     break;
   }
-- 
2.9.0

